Web Inspector: Rearrange Gradient Editor to be able to show the color component input fields

patrickangle opened the pull request:
#### 6d528bd10b503c55219feeb244cc8be89cd18ad8
<pre>
Web Inspector: Rearrange Gradient Editor to be able to show the color component input fields
<a href="https://bugs.webkit.org/show_bug.cgi?id=243352">https://bugs.webkit.org/show_bug.cgi?id=243352</a>
rdar://97803004

Reviewed by Devin Rousso.

Move the angle control above the color controls so that controls become more specific as you move from top to bottom of
the gradient editor. This also affords us the breathing room to show the color component fields at the bottom of the
color picker.

* Source/WebInspectorUI/UserInterface/Views/ColorPicker.js:
(WI.ColorPicker.prototype.async colorInputsWrapperElement):
(WI.ColorPicker.prototype._showColorComponentInputs):
(WI.ColorPicker.set enableColorComponentInputs): Deleted.
* Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
(.gradient-editor):
(.gradient-editor &gt; .gradient-type-select):
(.gradient-editor &gt; .gradient-slider):
(.gradient-editor &gt; .color-picker):
(.gradient-editor &gt; .gradient-angle):
(.gradient-editor.radial-gradient): Deleted.
(.gradient-editor.editing-color): Deleted.
(.gradient-editor.radial-gradient.editing-color): Deleted.
* Source/WebInspectorUI/UserInterface/Views/GradientEditor.js:
(WI.GradientEditor):

Canonical link: <a href="https://commits.webkit.org/252971@main">https://commits.webkit.org/252971@main</a>
</pre>

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GradientEditor.js:
```diff
@@ -115,6 +107,14 @@ WI.GradientEditor = class GradientEditor extends WI.Object
+
+        this._colorPicker = new WI.ColorPicker;
+        this._colorPicker.colorSquare.dimension = 195;
+        this._colorPicker.enableColorComponentInputs = false;
```
oops :P

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/ColorPicker.js:
```diff
@@ -161,12 +159,6 @@ WI.ColorPicker = class ColorPicker extends WI.Object
-    set enableColorComponentInputs(value)
-    {
-        this._enableColorComponentInputs = value;
-        this._element.classList.toggle("hide-inputs", !this._enableColorComponentInputs);
```
can we also get rid of the CSS `.color-picker.hide-inputs`?

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
```diff
@@ -24,25 +25,13 @@
  */
 
 .gradient-editor {
+    --gradient-editor-controls-width:  237px;
```
Style: should only have one space after `:`

NIT: I usually put CSS variables at the bottom of a declaration so that the actual things that affect style are shown first (similar to how we put member variables at the bottom of a `class` in C++)

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
```diff
@@ -24,25 +25,13 @@
-    padding-bottom: 15px;
-}
-
-.gradient-editor.editing-color {
```
it seems like we could also get rid of `"editing-color"` in the JS too?

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
```diff
@@ -24,25 +25,13 @@
 
 .gradient-editor > .gradient-type-select {
-    width: 237px;
+    width: var(--gradient-editor-controls-width);
```
would `width: 100%` not work here?

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
```diff
@@ -51,12 +40,12 @@
 
 .gradient-editor > .color-picker {
-    width: 238px;
+    width: calc(var(--gradient-editor-controls-width) + 1px);
```
ditto (:34)

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
```diff
@@ -65,12 +54,11 @@
 }
 
 .gradient-editor > .gradient-angle {
+    width: var(--gradient-editor-controls-width);
```
ditto (:34)

patrickangle committed:
Web Inspector: Rearrange Gradient Editor to be able to show the color component input fields
https://bugs.webkit.org/show_bug.cgi?id=243352
rdar://97803004

Reviewed by Devin Rousso.

Move the angle control above the color controls so that controls become more specific as you move from top to bottom of
the gradient editor. This also affords us the breathing room to show the color component fields at the bottom of the
color picker.

* Source/WebInspectorUI/UserInterface/Views/ColorPicker.js:
(WI.ColorPicker.prototype.async colorInputsWrapperElement):
(WI.ColorPicker.prototype._showColorComponentInputs):
(WI.ColorPicker.set enableColorComponentInputs): Deleted.
* Source/WebInspectorUI/UserInterface/Views/GradientEditor.css:
(.gradient-editor):
(.gradient-editor > .gradient-type-select):
(.gradient-editor > .gradient-slider):
(.gradient-editor > .color-picker):
(.gradient-editor > .gradient-angle):
(.gradient-editor.radial-gradient): Deleted.
(.gradient-editor.editing-color): Deleted.
(.gradient-editor.radial-gradient.editing-color): Deleted.
* Source/WebInspectorUI/UserInterface/Views/GradientEditor.js:
(WI.GradientEditor):

Canonical link: https://commits.webkit.org/252971@main
